timeless/views.py

```diff
@@ -30,86 +49,148 @@ def delete(self):
+        """
+        return self.context_object_list_name
+
+    def get_object_list(self):
+        """
+        Get the list of objects. If this method is not overwritten, then a
+        model variable must be declared, and it must have query.all().
+        """
+        if self.model is None:
+            raise NotImplementedError(f"{self.__class__.__name__} must define "
+                                      f"either 'model' or 'get_object_list()'")
+        return self.model.query.all()
+
+    def get_default_context(self):
+        """
+        Add the object list to the context.
+        """
+        context = super().get_default_context()
+        context[self.get_context_object_list_name()] = self.get_object_list()
+        return context
 
 
 class DetailView(GenericView):
     """
-    Example:
-        class TableView(DetailView):
-            def get(self, pk=None, *args, **kwargs):
-                if pk:
-                    table = models.Table.query.get(id)
-                    if not table:
-                        return redirect(url_for("tables.list"))
-                    form = forms.TableForm(request.form, instance=table)
-                else:
-                    form = forms.TableForm(request.form)
-                return render_template(
-                    "restaurants/tables/create_edit.html", form=form)
-
-            def post(self, pk=None, *args, **kwargs):
-                form = forms.TableForm(request.form)
-                if form.validate():
-                    form.save()
-                    return redirect(url_for("tables.list"))
-                return render_template(
-                    "restaurants/tables/create_edit.html", form=form)
+    A view that will display details in a template for a single object.
     """
+    context_object_name = "object"
+
+    def get_context_object_name(self):
+        """
+        Get context_object_name.
+        """
+        return self.context_object_name
+
     def get_object(self):
-        """ Method for fetching object from db"""
-        raise NotImplementedError()
+        """
+        Get the object. We don't make any assumptions, so this must be
+        overwritten by the subclass.
+        """
+        raise NotImplementedError(f"self.__class__.__name__ must define "
+                                  f"'get_object()'")
+
+    def get_default_context(self):
+        """
+        Add the object to the context.
+        """
+        context = super().get_default_context()
+        context[self.get_context_object_name()] = self.get_object()
+        return context
 
     def get(self, *args, **kwargs):
-        """ Get method implementation """
-        raise NotImplementedError()
-
-    def post(self, *args, **kwargs):
-        """ Post method implementation"""
-        raise NotImplementedError()
+        """
+        Set the object to an instance variable, then process as normal.
+        """
+        self.object = self.get_object()
```

Comment:
@bkuzmic hm, but this doesnt make sense - you have already passed `object` to context in `get_context` method. Looks like `get` implementation from `GenericView` is enough and it's not needed to override.

Reply:
@vryazanov You are right. I'll remove it.